nixos/modules/services/display-managers/cosmic-greeter.nix

```diff
@@ -50,14 +52,21 @@ in
       };
     };
 
+    systemd.tmpfiles.rules = [
+      "d /var/lib/cosmic-greeter 0750 cosmic-greeter cosmic-greeter -"
+      "d /run/cosmic-greeter 0755 cosmic-greeter cosmic-greeter -"
+    ];
+
     # The greeter user is hardcoded in `cosmic-greeter`
     users.groups.cosmic-greeter = { };
     users.users.cosmic-greeter = {
       description = "COSMIC login greeter user";
       isSystemUser = true;
       home = "/var/lib/cosmic-greeter";
-      createHome = true;
+      homeMode = "0750";
+      createHome = false;
```

Comment:
Have you tested to make sure this does still work with the `/var/lib/cosmic-greeter` removed? It's not empty on my machine; the `.cache`, `.config`, and `.local` folders are there, with the cache populated, `.config` containing actual configuration, and `.local` containing... something, it seems important (https://github.com/pop-os/cosmic-comp/issues/1040)